Security comms (#941)
elastic/docs-content: deploy-manage/security/self-setup.md

---

> navigation_title: "Self-managed security setup"

Author: thanks! yes, thanks for suggesting it. It turned out to make sense and to be like a `missing piece` before.

applies_to:
  deployment:
    self: ga
mapped_pages:
  - https://www.elastic.co/guide/en/elasticsearch/reference/current/security-basic-setup-https.html
---

# Set up security in self-managed deployments

There are two approaches to configuring security in self-managed deployments of the Elastic Stack:

## Automatic configuration [automatic-configuration]

Since version 8.0, {{es}} automatically enables security features on first startup when the node is not part of an existing cluster and none of the [incompatible settings](./self-auto-setup.md#stack-existing-settings-detected) have been explicitly configured.

> The automatic configuration::

Author: I've done the other way around, as the child file list was all in passive voice.

* Generates TLS certificates for the transport and HTTP layers

> * Applies security settings to elasticsearch.yml

Author: It means that the auto configuration process updates and modifies `elasticsearch.yml`, yes.

It's interesting because the automatic setup help users to learn the needed config, because the config ends up in `elasticsearch.yml`. Should we explain this better?

> * Applies security settings to elasticsearch.yml

Author: I've updated it to `Applies TLS configuration settings to `elasticsearch.yml`. Hoping it's clearer.

> * Applies security settings to elasticsearch.yml

Author: Oh... I know understand your concern :) the change should clearly solve that. Good point!

::::

Refer to [Automatic security setup](./self-auto-setup.md) for the complete procedure, including [cases where it may be skipped](./self-auto-setup.md#stack-skip-auto-configuration).

## Manual configuration [manual-configuration]

If you’re securing an existing unsecured cluster, or prefer to use your own TLS certificates, follow the manual approach. It involves enabling different layers of protection in sequence, depending on your cluster architecture and security requirements.

* **Start with [minimal security](set-up-minimal-security.md)**: Enables password-based authentication for built-in users and configures {{kib}} to connect using credentials. Suitable for single-node clusters, but not sufficient for production or multi-node clusters.

* **Then [configure transport TLS](./set-up-basic-security.md)**: Required for multi-node clusters running in [production mode](../deploy/self-managed/bootstrap-checks.md#dev-vs-prod-mode). Secures communication between nodes and prevents unauthorized nodes from joining the cluster.

> * **Finally, [configure HTTP TLS](set-up-basic-security-plus-https.md)**: Secures all client communications over HTTPS, including traffic between {{kib}} and {{es}}, and between browsers and {{kib}}. Recommended for all clusters, even single-node setups.

Author: Perfect, I'd like to share something else with you, because the process / numbered list is a bit `buggy` :)

Self-managed deployments support two approaches for the initial setup: automatic and manual. Note that securing {{kib}} always requires some manual configuration.

For guidance on configuring additional security features, refer to [](./secure-your-cluster-deployment.md).

## Automatic configuration [automatic-configuration]

Since version 8.0, {{es}} automatically enables security features on first startup when the node is not part of an existing cluster and none of the [incompatible settings](./self-auto-setup.md#stack-existing-settings-detected) have been explicitly configured.

The automatic configuration:

* Generates TLS certificates for the [transport and HTTP layers](./secure-cluster-communications.md#communication-channels)
* Applies TLS configuration settings to `elasticsearch.yml`
* Sets a password for the `elastic` superuser
* Creates an enrollment token to securely connect {{kib}} to {{es}}

This automatic setup is the quickest way to get started and ensures your cluster is protected by default.

> ::::{note}

Reviewer: ```suggestion
::::{warning}
```

The automatic configuration does not enable TLS on the {{kib}} HTTP endpoint. To encrypt browser traffic to {{kib}}, follow the steps in [](./set-up-basic-security-plus-https.md#encrypt-kibana-browser).

> ::::

Reviewer: I think we want this info here too:

https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/941/deploy-manage/security/secure-cluster-communications#encrypt-http-communication

Author: Added in the self-managed tab.

If you’re securing an existing unsecured cluster, or prefer to use your own TLS certificates, follow the manual approach. It involves enabling different layers of protection in sequence, depending on your cluster architecture and security requirements.

1. **[Set up minimal security](set-up-minimal-security.md)**: Enables password-based authentication for built-in users and configures {{kib}} to connect using credentials. Suitable for single-node clusters, but not sufficient for production or multi-node clusters.

2. **[Configure transport TLS](./set-up-basic-security.md)**: Required for multi-node clusters running in [production mode](../deploy/self-managed/bootstrap-checks.md#dev-vs-prod-mode). Secures communication between nodes and prevents unauthorized nodes from joining the cluster.

3. **[Configure HTTP TLS](set-up-basic-security-plus-https.md)**: Secures all client communications over HTTPS, including traffic between {{kib}} and {{es}}, and between browsers and {{kib}}. Recommended for all clusters, even single-node setups.

Each step builds on the previous one. For production environments, it’s strongly recommended to complete all three.

## Kibana security configuration

> Refer to [](./using-kibana-with-security.md) for detailed guidance on the following security topics:

Reviewer: ```suggestion
Refer to [](./using-kibana-with-security.md) to learn how to implement the following security best practices for {{kib}}:
```

# Set up security in self-managed deployments

There are two approaches to configuring security in self-managed deployments of the {{stack}}:

## Automatic configuration [automatic-configuration]

Since version 8.0, {{es}} automatically enables security features on first startup when the node is not part of an existing cluster and none of the [incompatible settings](./self-auto-setup.md#stack-existing-settings-detected) have been explicitly configured.

> The automatic configuration::

Reviewer: ```suggestion
The automatic configuration:
```
